src/utils/api/withAuthorization.ts

```diff
@@ -0,0 +1,28 @@
+import {
+	APIGatewayProxyEvent,
+	APIGatewayProxyHandler,
+	APIGatewayProxyResult,
+	Callback,
+	Context,
+} from 'aws-lambda';
+import { validateAuthorizationToken } from '@utils/api/apiAuth';
+
+export function withAuthorization(handler: APIGatewayProxyHandler): APIGatewayProxyHandler {
+	return async (
+		event: APIGatewayProxyEvent,
+		context: Context,
+		callback: Callback<APIGatewayProxyResult>
+	): Promise<APIGatewayProxyResult> => {
+		const authorizationValidationResult = validateAuthorizationToken(event);
```

Comment:
I think this is one layer of abstraction too many. You should pull the internals of this function inline here IMO